pkg/rlptools/wasm/utils.go

```diff
@@ -287,3 +297,130 @@ func dataFromLimit(limitIdentifier string, limit *kuadrantv1beta2.Limit) (data [
 
 	return data
 }
+
+func routeFromRLP(ctx context.Context, t *kuadrantgatewayapi.TopologyIndexes, rlp *kuadrantv1beta2.RateLimitPolicy, gw *gatewayapiv1.Gateway) (*gatewayapiv1.HTTPRoute, error) {
+	logger, err := logr.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	route := t.GetPolicyHTTPRoute(rlp)
+
+	if route == nil {
+		// The policy is targeting a gateway
+		// This gateway policy will be enforced into all HTTPRoutes that do not have a policy attached to it
+
+		// Build imaginary route with all the routes not having a RLP targeting it
+		untargetedRoutes := t.GetUntargetedRoutes(gw)
```

Comment:
I was banging my head against the wall for 20 min or so, trying to find where in the code we "fixed" the imaginary route thing for RLPs. It turns out we haven't. 🙄

Leaving this comment here then only so we remember to fix the issue here as well whenever it is addressed with #688.

IOW, this LOC is wrong, but consistently wrong with all the other places where we made the same mistake, making it kind of right, if it makes any sense 😵‍💫

Reply:
good point. Thanks for that. That will ease the process of rebasing when we fix that in `main`. With a little bit of luck, we can merge this into `main` before we fix it and then no rebase conflicts.